### ISSUE TITLE
[Paginator] Mobile version of paginator has wrong value of `aria-label`

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3629]
### 😯 Current Behavior

Select used to change pages in mobile version of paginator has `aria-label` that says `X, page`, where X is current page number - 1.

Because of that, when user place focus on paginator with 1 page selected screen reader announces this element as: '1, 0, page, listbox, pop up collapsed, combo box'. This may be confusing for users.

https://github.com/user-attachments/assets/b2701799-c134-4115-8dbe-98f354a9b73b

<img width="332" height="111" alt="Image" src="https://github.com/user-attachments/assets/6ea0facf-b9cc-40d7-8351-499037d57674" />

### 🤔 Expected Behavior/possible solution?

`aria-label` for that select should be changed so it no longer include 'number of current page -1'.

`aria-label='Page'` will be sufficient. Then paginator will be announced as '1, Page, listbox, pop up collapsed, combo box'.

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/navigation-paginator-paginator--mobile
2. With screen reader turned on place keyboard focus on select
3. Verify if screen reader announce any other number than current page number

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/navigation-paginator--connected

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

0.2.8

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
